src/frontend/src/scheduler/distributed/query.rs

```diff
@@ -415,12 +417,14 @@ impl QueryRunner {
     }
 
     async fn all_children_scheduled(&self, stage_id: &StageId) -> bool {
-        for child in self.query.stage_graph.get_child_stages_unchecked(stage_id) {
-            if !self.stage_executions[child].is_scheduled().await {
-                return false;
-            }
-        }
-        true
+        self.query
+            .stage_graph
+            .get_child_stages_unchecked(stage_id)
+            .iter()
+            .map(|child| self.stage_executions[child].is_scheduled())
```

Comment:
`is_scheduled` only acquires a tokio RwLock. Does it have any improvement to make it concurrent?

Reply:
+1, I think this doesn't help much here.